lib/fritzbox/smarthome/actor.rb

```diff
@@ -4,36 +4,64 @@ class Actor < Resource
       include ActiveModel::Model
 
       attr_accessor \
-        :id,
-        :type,
-        :ain,
-        :present,
-        :name,
-        :manufacturer,
-        :group_members
+      :id,
+      :type,
+      :ain,
+      :present,
+      :name,
+      :manufacturer,
+      :group_members
+
+      ResourceNotFound = Class.new(RuntimeError)
 
       class << self
         def all(types: ['group', 'device'])
-          response = get(command: 'getdevicelistinfos')
-          xml = nori.parse(response.body)
+          xml = parse(get(command: 'getdevicelistinfos'))
+
           Array.wrap(types.map { |type| xml.dig('devicelist', type) }.flatten).compact.map do |data|
             klass = Actor.descendants.find { |k| k.match?(data) } || Actor
             self.in?([klass, Actor]) ? klass.new_from_api(data) : nil
           end.compact
         end
 
+        def find_by!(id: nil, ain: nil)
+          raise ArgumentError, 'Find via :id is not implmentet yet' if id.present?
```

Comment:
```suggestion
          raise ArgumentError, 'Find via :id is not implemented yet' if id.present?
```

But what would an id be? I would think the AIN already is a unique identifier

Reply:
This one here:

https://github.com/klausmeyer/fritzbox-smarthome/blob/aca8da316bc48f9442f5121fa0d47258ee439ec3/lib/fritzbox/smarthome/actor.rb#L27

It's returned as a numeric ID whereas the AIN looks like something based on the serial number or hardware address.

Thanks for pointing me to the typo - but as mentioned above I guess I'll remove that part completely to simplify.